nncf/quantization/layers.py

```diff
@@ -385,38 +433,41 @@ def broadcast_initialized_params(self, src: int = 0):
         distributed.broadcast(self.signed_tensor, src=src)
 
-    def run_export_quantization(self, x: torch.Tensor):
-        with no_jit_trace():
-            input_range = abs(self.scale) + self.eps
-            # todo: take bias into account during input_low/input_high calculation
-            input_low = input_range * self.level_low / self.level_high
-            input_high = input_range
-
-            if self._export_mode == QuantizerExportMode.ONNX_QUANTIZE_DEQUANTIZE_PAIRS:
-                y_scale, y_zero_point = get_scale_zp_from_input_low_input_high(self.level_low,
-                                                                               self.level_high,
-                                                                               input_low,
-                                                                               input_high)
+    def _get_input_low_input_high(self, scale, level_low, level_high, eps):
+        input_range = abs(scale) + eps
+        input_low = input_range * level_low / level_high
+        input_high = input_range
+        return input_low, input_high
 
-        if self._export_mode == QuantizerExportMode.ONNX_QUANTIZE_DEQUANTIZE_PAIRS:
-            if self.per_channel:
-                if torch.allclose(y_scale - y_scale[0], torch.zeros_like(y_scale)) and torch.allclose(
-                        y_zero_point - y_zero_point[0], torch.zeros_like(y_zero_point)):
-                    y_scale, y_zero_point = y_scale[0], y_zero_point[0]
-                    return ExportQuantizeToONNXQuantDequant.apply(x, y_scale, y_zero_point)
-                raise RuntimeError("PyTorch v1.5.0 export to ONNX using QuantizeLinear-DequantizeLinear "
-                                   "doesn't support per channel quantization")
-            return ExportQuantizeToONNXQuantDequant.apply(x, y_scale, y_zero_point)
-        if self._export_mode == QuantizerExportMode.FAKE_QUANTIZE:
-            return ExportQuantizeToFakeQuantize.apply(x, self.levels, input_low, input_high, input_low, input_high)
-        raise RuntimeError
+    def _prepare_export_quantization(self, x: torch.Tensor):
+        with no_jit_trace():
+            input_low, input_high = self._get_input_low_input_high(self.scale,
+                                                                   self.level_low,
+                                                                   self.level_high,
+                                                                   self.eps)
+            level_low = self.level_low
+            level_high = self.level_high
+            if self._is_applied_saturation_fix:
+                if self.scale_shape[0] > 1:
+                    for i in range(self.scale_shape[0]):
+                        x[i] = torch.clamp(x[i], min=input_low[i].item(), max=input_high[i].item())
+                else:
+                    x.data = torch.clamp(x, min=input_low.item(), max=input_high.item())
+                level_low = 2 * self.level_low
```

Comment:
Is the formula correct for the unsigned cases?

Reply:
I rechecked the formula once again. It has to work. There are two possible cases how levels are changed during export:
1) Signed quantization.
`prev_level_low = -64  ----> new_level_low = -128`
`prev_level_high = 63  -----> new_level_high = 127
`
2) Unsigned quantization.
`prev_level_low = 0 -> new_level_low = 0`
`prev_level_high = 127 -> new_level_high = 255`

The formula:
`level_low = 2 * level_low`
`level_high = 2 * level_high + 1`

For both cases the formula works
1) Signed 
`level_low = 2 * -64 = -128`
`level_high = 2 * 63 + 1 = 127`
2) Unsigned 
`level_low = 2 * 0 = 0`
`level_high = 2 * 127 + 1 = 255`

Also it works for various number of quantizer bits

Reply:
Thanks, for the detailed answer.